carparkmanager-persistence/src/main/java/cz/muni/fi/pa165/carparkmanager/persistence/entity/Employee.java

```diff
@@ -0,0 +1,139 @@
+
+/**
+ *
+ * @author xbonco1
+ */
+@Entity
+public class Employee {
+    @Id
+    @GeneratedValue(strategy = GenerationType.TABLE)
+    @Column(name="EMPLOYEE_ID")
+    private Long id;
+    
+    @NotNull
+    @Column(nullable = false)
+    private String firstname;
+
+    @NotNull
+    @Column(nullable = false)
+    private String surname; 
+    
+    @NotNull
+    @Temporal(TemporalType.DATE)
+    private Date birthDate;
+
+    @NotNull
+    @Enumerated
+    private ClassificationOfEmployee classification;    
+
+    public Employee() {
+        
+    }
+    
+    public Employee(Long id, String firstname, String surname, Date birthDate, ClassificationOfEmployee classification) {
+        this.id = id;
+        this.firstname = firstname;
+        this.surname = surname;
+        this.birthDate = birthDate;
+        this.classification = classification;         
+    }
+    
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+    
+    public String getFirstname() {
+        return firstname;
+    }
+    
+    public void setFirstname(String firstname) {
+        this.firstname = firstname;
+    }
+
+    public String getSurname() {
+        return surname;
+    }
+
+    public void setSurname(String surname) {
+        this.surname = surname;
+    }
+
+    public Date getBirthDate() {
+        return birthDate;
+    }
+
+    public void setBirthDate(Date dateOfBirth) {
+        this.birthDate = dateOfBirth;
+    }
+
+    public ClassificationOfEmployee getClassification() {
+        return classification;
+    }
+
+    public void setClassification(ClassificationOfEmployee classification) {
+        this.classification = classification;
+    }
+    
+    @Override
+    public int hashCode() {
+        int hash = 3;
+        hash = 73 * hash + Objects.hashCode(this.id);
+        hash = 73 * hash + Objects.hashCode(this.birthDate);
+        hash = 73 * hash + Objects.hashCode(this.firstname);
+        hash = 73 * hash + Objects.hashCode(this.surname);
+        hash = 73 * hash + Objects.hashCode(this.classification);
+
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (!(obj instanceof Employee)) {
+            return false;
+        }
+        
+        final Employee other = (Employee) obj;
+        if (!Objects.equals(this.id, other.getId())) {
+            return false;
+        }
+        if (!Objects.equals(this.firstname, other.getFirstname())) {
+            return false;
+        }
+        if (!Objects.equals(this.surname, other.getSurname())) {
+            return false;
+        }
+        if (!Objects.equals(this.birthDate, other.getBirthDate())) {
+            return false;
+        }
+        if (this.classification != other.getClassification()) {
```

Comment:
posledny if je redundantny - return this.classification == other.getClassification();

Reply:
mas pravdu, ale prišlo mi to prehladnejšie - mohol som to zabit cele aj do jedneho riadku :D

Reply:
to mi ponuklo IDE :D ale hej je to jedno